test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java

```diff
@@ -256,9 +256,9 @@ private void registerTxnSubmitted(HapiApiSpec spec) throws Throwable {
 					TransactionID id = spec.registry().getTxnId(name);
 					builder.setTransactionID(id);
 				});
-				if (nonce != NO_NONCE) {
-					builder.getTransactionIDBuilder().setNonce(ByteString.copyFrom(nonce));
-				}
+			}
+			if (nonce != NO_NONCE) {
+				builder.getTransactionIDBuilder().setNonce(ByteString.copyFrom(nonce));
 			}
```

Comment:
I have one question. If `nonce` field is meant for scheduled transaction only, should these 3 lines be moved to the  
  ` if (omitTxnId) { ... }` block?

Reply:
You're right; I think I put it here instead of the `if (omitTxnId) ...` block to make it easier to construct an invalid transaction. 🤔 

Either one is vastly superior to where it was before---which was a clear bug haah.

Reply:
I see. It makes sense now if we also want to create different test scenarios.